modules/xmpp/JingleSessionPC.js

```diff
@@ -1808,11 +1816,20 @@ export default class JingleSessionPC extends JingleSession {
             sdp: remoteSdp
         });
 
-        if (this.isInitiator) {
-            return this._initiatorRenegotiate(remoteDescription);
-        }
+        const promise = this.isInitiator
+            ? this._initiatorRenegotiate(remoteDescription)
+            : this._responderRenegotiate(remoteDescription);
+
+        return promise.then(() => {
+            // Publish the codec info to the other endpoints in the conference if it has changed
+            // as a result of the renegotiation (only on current active session).
+            const codec = this.peerconnection.getConfiguredVideoCodec();
```

Comment:
When participant A supports VP9, but then comes participant B who doesn't, then is it the case that both participants A and B will now be advertising their codec as VP8 after participant A adopts? Is that intentional to never switch back?

Reply:
The switch back happens here when participant B leaves or all participants that don't have the preferred codec leave -https://github.com/jallamsetty1/lib-jitsi-meet/blob/codec-preference/modules/RTC/CodecSelection.js#L164.
The idea was to keep the current codec in presence so that we can use that for bitrate adaptation as a next step.